shared/reporter/collect.go

```
@@ -0,0 +1,222 @@
+// permissive: this reporter only ever looks up a handful of known series, and a
+// line it cannot parse must not cost us the ones it can.
+func parseMetrics(reader io.Reader) []sample {
+	var samples []sample
+	scanner := bufio.NewScanner(reader)
+	scanner.Buffer(make([]byte, 0, 64*1024), 1024*1024)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		name, remainder := line, ""
```

golangci-lint (2.12.2)

[error] ineffectual assignment to name

(ineffassign)